overlord/snapstate/snapstate.go

```diff
@@ -75,30 +74,17 @@ func isParallelInstallable(snapsup *SnapSetup) error {
 }
 
 func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int, fromChange string) (*state.TaskSet, error) {
+	// NB: ideally deviceCtx should not need to propagate here!
```

Comment:
Not sure I get this comment - I don't see deviceCtx used in this function?

Reply:
Would:

// NB:  we should strive not to need or propagate deviceCtx here,  the resulting effects/changes were not pleasant at one point

be clearer?